src/main/java/apoc/algo/pagerank/PageRankArrayStorageParallelCypher.java

```diff
@@ -0,0 +1,324 @@
+        relationshipWeight = new int[totalRelationships];
+
+        for (int i = 0; i < totalRelationships; i++) {
+            relationshipTarget[i] = -1;
+            relationshipWeight[i] = -1;
+        }
+        this.relCount = totalRelationships;
+        calculateChunkIndices();
+
+        // We have degrees for all the nodes at the point.
+        result = db.execute(relCypher);
+        while(result.hasNext()) {
+            Map<String, Object> res = result.next();
+            int source = ((Long) res.get("source")).intValue();
+            int target = ((Long) res.get("target")).intValue();
+            int weight = ((Long) res.getOrDefault("weight", 1)).intValue();
+
+            int logicalSourceIndex = getNodeIndex(source);
+            int logicalTargetIndex = getNodeIndex(target);
+            int chunkIndex = sourceChunkStartingIndex[logicalSourceIndex];
+            while(relationshipTarget[chunkIndex] != -1) {
+                chunkIndex++;
+            }
+            relationshipTarget[chunkIndex] = logicalTargetIndex;
+            relationshipWeight[chunkIndex] = weight;
+
+            previousPageRanks[logicalSourceIndex] = 0;
+            previousPageRanks[logicalTargetIndex] = 0;
+            pageRanks[logicalTargetIndex] = 0;
+        }
+
+        result.close();
+    }
+
+    @Override
+    public void compute(
+            int iterations,
+            RelationshipType... relationshipTypes)
+    {
+        for (int iteration = 0; iteration < iterations; iteration++) {
+            startIteration();
+            iterate();
+        }
+
+        previousPageRanks = null;
+        sourceDegreeData = null;
+        sourceWeightData = null;
+        sourceChunkStartingIndex = null;
+        relationshipTarget = null;
+        relationshipWeight = null;
+    }
+
+    private void iterate() {
+
+        for (int i = 0; i < nodeCount; i++) {
+            int chunkIndex = sourceChunkStartingIndex[i];
+            int degree = sourceDegreeData[i];
+
+            for (int j = 0; j < degree; j++) {
+                int source = i;
+                int target = relationshipTarget[chunkIndex + j];
+                int weight = relationshipWeight[chunkIndex + j];
+
+                int oldValue = pageRanks[target];
+                int newValue =  oldValue + weight * previousPageRanks[source];
+                pageRanks[target] = newValue;
+            }
+        }
+    }
+
+    private void startIteration()
+    {
+        for (int node = 0; node < nodeCount; node++) {
+            int weightedDegree = sourceWeightData[node];
+
+            if (weightedDegree == -1) {
+                continue;
+            }
+            int prevRank = pageRanks[node];
+            previousPageRanks[node] =  toInt(ALPHA * toFloat(prevRank) / weightedDegree);
+            pageRanks[node] = ONE_MINUS_ALPHA_INT;
+        }
+    }
+
+    public void writeResultsToDB() {
+        writeBackResults(db, this);
```

Comment:
Yep. Would make this static too.
Initially I thought I should move this function to PagerankUtils.java. Would that be okay?